HeterogeneousCore/SonicTriton/interface/TritonClient.h

```diff
@@ -0,0 +1,93 @@
+#ifndef HeterogeneousCore_SonicTriton_TritonClient
+#define HeterogeneousCore_SonicTriton_TritonClient
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "HeterogeneousCore/SonicCore/interface/SonicClientSync.h"
+#include "HeterogeneousCore/SonicCore/interface/SonicClientPseudoAsync.h"
+#include "HeterogeneousCore/SonicCore/interface/SonicClientAsync.h"
+
+#include <vector>
+#include <string>
+#include <exception>
+
+#include "request_grpc.h"
+
+namespace ni = nvidia::inferenceserver;
+namespace nic = ni::client;
+
+template <typename Client>
+class TritonClient : public Client {
+public:
+  struct ServerSideStats {
+    uint64_t request_count_;
+    uint64_t cumul_time_ns_;
+    uint64_t queue_time_ns_;
+    uint64_t compute_time_ns_;
+  };
+
+  //constructor
+  TritonClient(const edm::ParameterSet& params);
+
+  //helper
+  bool getResults(const std::unique_ptr<nic::InferContext::Result>& result);
+
+  //accessors
+  unsigned nInput() const { return nInput_; }
+  unsigned nOutput() const { return nOutput_; }
+  unsigned batchSize() const { return batchSize_; }
+  bool verbose() const { return verbose_; }
+  void setBatchSize(unsigned bsize) { batchSize_ = bsize; }
+
+  //for fillDescriptions
+  static void fillPSetDescription(edm::ParameterSetDescription& iDesc) {
+    edm::ParameterSetDescription descClient;
+    descClient.add<unsigned>("nInput");
+    descClient.add<unsigned>("nOutput");
+    descClient.add<unsigned>("batchSize");
+    descClient.add<std::string>("address");
+    descClient.add<unsigned>("port");
+    descClient.add<unsigned>("timeout");

```

Comment:
I think some of the parameters (e.g. `address`, `port`, `timout`, ...) should be `untracked`, since they are not supposed to impact the physics results

Reply:
I'm willing to make this change, but it's not impossible that the choice of server could impact the physics results. For example, floating point math might give slightly different results.

Reply:
Honestly, that would just be a limitation of the implementation - like we can get slightly different rersults due to different CPU vendors.

But if this is ever used in production, I don't think one can assume that a single server should be able to handle the requests for a whole dataset: for example, parts of a dataset could be processed in different sites, which then would likely need to use different servers.

Reply:
Based on this idea, I think the only tracked parameters should be: nInput, nOutput, modelName, modelVersion. I will change the rest to untracked.

Reply:
(thinking out loud) On the other hand, we are tracking in the provenance the choices of `SwitchProducer`, and we are going to add tracking of the chosen "vector architecture" (the recent scram feature). Further on the other hand, we do not track the exact CPU type (e.g. Intel vs AMD for non-standard fast math) nor `$SCRAM_ARCH` (e.g. amd64 vs arm vs ppc), and one could easily argue the exact server to be similar to either of those.

Anyway, keeping them untracked is fine for me now.